IGInterfaceDataTable/WKInterfaceTable+IGInterfaceDataTable.m

```diff
@@ -341,6 +381,39 @@ - (void)insertRowsAtIndexPaths:(NSArray *)indexPaths withRowType:(NSString *)row
   }
 }
 
+- (void)beginUpdates {
+  self.updating = true;
+  self.rowIndexesToAdd = [[NSMutableArray alloc] init];
+  self.rowTypesToAdd = [[NSMutableDictionary alloc] init];
+}
+
+- (void)endUpdates {
+  if(!self.updating) {
+    return;
+  }
+
+  [self.rowIndexesToAdd sortUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+    NSInteger r1 = [obj1 row];
+    NSInteger r2 = [obj2 row];
+    if (r1 > r2) {
+      return (NSComparisonResult)NSOrderedDescending;
+    }
+    if (r1 < r2) {
+      return (NSComparisonResult)NSOrderedAscending;
+    }
+    return (NSComparisonResult)NSOrderedSame;
+  }];
+
+  self.updating = false;
+
+  [self.rowIndexesToAdd enumerateObjectsUsingBlock:^(NSIndexPath *indexPath, NSUInteger idx, BOOL *stop) {
+    [self insertRowsAtIndexPaths:@[indexPath] withRowType:[self.rowTypesToAdd objectForKey:indexPath]];
```

Comment:
We're using subscripting throughout the project, let's keep it consistent here: `self.rowTypesToAdd[indexPath]`